### PR TITLE
Storage account lockdown

### PR DESCRIPTION
**Pull request by mjudeikis**
### Which issue this PR addresses:

Enables storage account lockdown.
Quite a big change so early reviews would be good. Let me know if people want separate review call for this one. 

### What this PR does / why we need it:

This pr will do few things to make storage accounts side more secure:
1. Deploy all storage accounts encrypted (fedramp) (done, tests missing)
2. Add firewalls rules for RP and customer subnets (done, tests missing)
3. Add controller to make sure firewall rules and service endpoints are configures right (if users adds new subnet) 
4. Remove validation for service endpoint in favor of install step (done, tests missing)
5. Add AdminPutOrPatch to migrate existing infrastrucutre to locked setup (all except encryption as its can't be done)

Split: 
https://github.com/Azure/ARO-RP/pull/1663 - Subnet related changes in the operator mostly 
https://github.com/Azure/ARO-RP/pull/1664 - client updates 

TODO:
- [x] Refactor subnet controller to be extendable
- [x] Add firewall reconcile
- [ ] Add prepopulate storage account name in db for old clusters
- [x] reconcile old storage accounts 
- [ ] Add controller tests
- [ ] Add ARM feature flag for this so we can test 

### Test plan for issue:

Unit tests and mocks

-->

### Is there any documentation that needs to be updated for this PR?

Yes

### CODE REVIEW

File: pkg/cluster/storegaaccounts.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
+package cluster
```
Comment (hawkowl) [inline pkg/cluster/storegaaccounts.go:1]: "`storegaaccounts`" is a typo, I think

File: pkg/cluster/storegaaccounts.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+
+	"github.com/Azure/ARO-RP/pkg/util/arm"
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+// enableServiceEndpointstEndpoints should enable Microsoft.Storage endpoints on
+// subnets for storage account access
+func (m *manager) enableServiceEndpointstEndpoints(ctx context.Context) error {
```
Comment (hawkowl) [inline pkg/cluster/storegaaccounts.go:20]: I don't like this function name because it looks like a typo even though it probably isn't

Comment (mjudeikis) [reply pkg/cluster/storegaaccounts.go:20]: it is :) good catch

**Comment by mjudeikis** [comment]
Gonna be splitting

**Pull request closed without merging**